### PR TITLE
Add p4api

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages
- [ ] Build number is 0
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/p4api/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - test -f ${PREFIX}/lib/libp4script_c.a   # [unix]
+    - test -f ${PREFIX}/lib/librpc.a          # [unix]
+    - if not exist %PREFIX%\\Library\\LIB\\libclient.a       # [win]
+    - if not exist %PREFIX%\\Library\\LIB\\libp4script.a     # [win]
+    - if not exist %PREFIX%\\Library\\LIB\\libp4script_c.a   # [win]
+    - if not exist %PREFIX%\\Library\\LIB\\librpc.a          # [win]
+
+about:
+  home: http://www.perforce.com
+  license: Freely Distributable
```
Comment (marcelotrevisani) [inline recipes/p4api/meta.yaml]: I am not sure what is the right license I should write here

File: recipes/p4api/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  sha256: 342bfc820a988db11002ff1804f404796f06b214ba7630d6d8a3593aa564b7bf           # [win]
+
+build:
+  number: 0
+  script:
+    - cp -r include ${PREFIX}    # [unix]
+    - cp -r lib ${PREFIX}        # [unix]
+    - xcopy /E include %PREFIX%  # [win]
+    - xcopy /E lib %PREFIX%      # [win]
+  skip: True  # [win]
```
Comment (marcelotrevisani) [inline recipes/p4api/meta.yaml]: Skipping windows for now. I will add it later

**Comment by marcelotrevisani** [comment]
@conda-forge/staged-recipes , couldn't you please review it?

File: recipes/p4api/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/p4api/meta.yaml]: Looks like a BSD-2-Clause license, no?

Comment (marcelotrevisani) [reply recipes/p4api/meta.yaml]: Indeed! Looks like it, I'm going to change it
Thanks! :)

**Comment by isuruf** [comment]
Can you build this package from source?

**Comment by marcelotrevisani** [comment]
> Can you build this package from source?

Unfortunately, I can't  :(
This company just share the API already compiled

**Comment by isuruf** [comment]
What kind of information is there about these static libraries? Are they compatible with the compilers and runtimes we use?

File: recipes/p4api/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+requirements:
+  run:
+    - openssl
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libclient.a            # [unix]
+    - test -f ${PREFIX}/lib/libp4script.a          # [unix]
+    - test -f ${PREFIX}/lib/libp4script_c.a        # [unix]
+    - test -f ${PREFIX}/lib/librpc.a               # [unix]
```
Comment (isuruf) [inline recipes/p4api/meta.yaml]: Isn't this a separate project?

**Comment by marcelotrevisani** [comment]
> What kind of information is there about these static libraries? Are they compatible with the compilers and runtimes we use?
I know that the .net api needs vs 2013
for unix I know that is abi compatible with recent compilers

File: recipes/p4api/meta.yaml (hunk 3 above)
Comment (marcelotrevisani) [reply recipes/p4api/meta.yaml]: Sorry, I didn't get you, could you please explain to me?

Comment (isuruf) [reply recipes/p4api/meta.yaml]: Is librpc.a part of p4api?

Comment (marcelotrevisani) [reply recipes/p4api/meta.yaml]: Yes, it is

**Comment by isuruf** [comment]
Sources are in https://swarm.workshop.perforce.com/projects/perforce_software-p4/files/2018-2

**Comment by marcelotrevisani** [comment]
> Sources are in https://swarm.workshop.perforce.com/projects/perforce_software-p4/files/2018-2

Ooowww, I searched for it a lot and I didn't find the source. Thanks, I will create the build scripts for it then
:D

**Comment by marcelotrevisani** [comment]
It needs ``Jam`` to compile it, so, I am creating a recipe for it

**Comment by marcelotrevisani** [comment]
Related to 
https://github.com/conda-forge/staged-recipes/pull/8540

**Comment by jakirkham** [comment]
It looks like `Jam` was added. Would it be possible to explore building from source now or are there other issues?

**Comment by marcelotrevisani** [comment]
> It looks like `Jam` was added. Would it be possible to explore building from source now or are there other issues?

Sure, I will try to compile it from source

File: recipes/p4api/meta.yaml (hunk 3 above)
Comment (marcelotrevisani) [reply recipes/p4api/meta.yaml]: I was checking and you are right, actually ``librpc`` is not a separated packaged and p4 does not list it as a dependency.

**Comment by marcelotrevisani** [comment]
It depends on ``librpc`` and I don't know how to set the script to use conda compilers
I don't have much experience with ``jam``

**Comment by scopatz** [comment]
something is causing the linter to fail

**Pull request closed without merging**